code/datums/weather/weather.dm

```diff
@@ -189,6 +165,23 @@
 	SSweather.processing -= src
 	update_areas()
 
+// handles sending all alerts
+/datum/weather/proc/send_alert(alert_msg, alert_sfx)
+	for(var/z_level in impacted_z_levels)
+		for(var/mob/player as anything in SSmobs.clients_by_zlevel[z_level])
+			if(!can_get_alert(player))
+				continue
+			if(alert_msg)
+				to_chat(player, alert_msg)
+			if(alert_sfx)
+				SEND_SOUND(player, sound(alert_sfx))
+
+// the checks for if a mob should recieve alerts, returns TRUE if can
+/datum/weather/proc/can_get_alert(mob/player)
+	var/turf/mob_turf = get_turf(player)
+	if(mob_turf)
+		return TRUE
```

Comment:
```suggestion
	return !isnull(mob_turf)
```

You should always explicitly return values, your old code was going to return null on failure, which is incorrect.